src/mbgl/gl/attribute.hpp

```diff
@@ -95,5 +95,36 @@ class AttributeLocations<TypeList<As...>> final {
     }
 };
 
+template <class>
+class AttributeKey;
+
+constexpr auto attributeDefinePrefix() {
+    return "#define HAS_UNIFORM_u_";
+}
+
+template <class... As>
+class AttributeKey<TypeList<As...>> final {
+public:
+    static_assert(sizeof...(As) <= 32, "attribute count exceeds 32");
+
+    static uint32_t compute(const gfx::AttributeBindings<TypeList<As...>>& bindings) {
+        uint32_t value = 0;
```

Comment:
nit: 0u

Reply:
why?

Reply:
`0` is `int`, not `unsigned`.. however compiler optimizes it out anyway, so this point is optional too